### PR TITLE
Generalize system setting and update code style

### PR DESCRIPTION
**Pull request by ChunYen-Chen**

### CODE REVIEW

File: src/configure.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def _get_option_tuples(self, option_string):
 
         # shouldn't ever get here
         else:
-            self.error(_('unexpected option string: %s') % option_string)
+            self.error(_("unexpected option string: %s") % option_string)
```
Comment (technic960183) [inline src/configure.py:150]:
```suggestion
            self.error(("unexpected option string: %s") % option_string)
```
Is `_` intentionally? Static checker yell at it said that it is not defined.

File: src/configure.py [hunk 2]
```diff
@@ -760,14 +769,50 @@ def load_config( config ):
...
+        lines = f.readlines()
+
+    for line in lines:
+        tokens = line.strip().split()
+        if len(tokens) == 0: continue      # empty line
+        if tokens[0][0] == "#": continue   # skip comment line
+        try:
+           sys_setting[tokens[0]] = tokens[1]
+        except:
+           sys_setting[tokens[0]] = None
```
Comment (technic960183) [inline src/configure.py:811]:
```suggestion
        if len(tokens) >= 2:
           sys_setting[tokens[0]] = tokens[1]
        else:
           sys_setting[tokens[0]] = None
```
I think in general we should avoid catch with non-explicitly exception.
Is there other case that will fail the suggestion?

File: src/configure.py [hunk 3]
```diff
@@ -760,14 +769,50 @@ def load_config( config ):
...
+    setting_file = None
 
+    if os.path.isfile( GAMER_GLOBAL_SETTING ): setting_file = GAMER_GLOBAL_SETTING
+    if os.path.isfile( GAMER_LOCAL_SETTING  ): setting_file = GAMER_LOCAL_SETTING
+
+    if setting_file is None:
+        LOGGER.info("System setting file not detected.")
+        return sys_setting
+    else:
+        LOGGER.info("Using %s as setting file."%(setting_file))
```
Comment (technic960183) [inline src/configure.py:799]:
```suggestion
    if   os.path.isfile( GAMER_LOCAL_SETTING  ): setting_file = GAMER_LOCAL_SETTING
    elif os.path.isfile( GAMER_GLOBAL_SETTING ): setting_file = GAMER_GLOBAL_SETTING
    else:
        LOGGER.info("System setting file not detected.")
        return sys_setting

    LOGGER.info("Using %s as setting file."%(setting_file))
```
Not necessary, but make the fall back logic cleaner.

**Review by technic960183 — COMMENTED** [review]
If you agree, I can modify it as my suggestion. Thanks!

File: src/configure.py (hunk 3 above)
Comment (technic960183) [reply src/configure.py:799]: In a more pythonic style, we can remove `setting_file = None` at the beginning.
But this violate the variable life cycle rule in `c/c++`.
Should we do this file in a pythonic flavor or a c/c++ flavor?

File: src/configure.py (hunk 1 above)
Comment (ChunYen-Chen) [reply src/configure.py:150]: I guess so. The reference of this line is at [here](https://github.com/python/cpython/blame/43709d5d5424725ec061de2c12c3761affde69ef/Lib/argparse.py#L2321)

File: src/configure.py (hunk 3 above)
Comment (ChunYen-Chen) [reply src/configure.py:799]: Agree. Let's remove `setting_file = None`.

File: src/configure.py (hunk 2 above)
Comment (ChunYen-Chen) [reply src/configure.py:811]: Agree, `if else` should be enough.
Can you also update these [lines](https://github.com/technic960183/gamer-fork-ymhsu/blob/b0573cee7917fe0c3f0d676f4f89aba1434360a8/src/configure.py#L760-L763) accordingly?

Comment (technic960183) [reply src/configure.py:811]: Sure.

File: src/configure.py (hunk 1 above)
Comment (technic960183) [reply src/configure.py:150]: I think we should remove it. As `_` is an alias from `gettext` module that we don't use at here.
([Ref](https://github.com/python/cpython/blame/43709d5d5424725ec061de2c12c3761affde69ef/Lib/argparse.py#L92))

Comment (ChunYen-Chen) [reply src/configure.py:150]: Agree. Let's remove it.

**Review by technic960183 — COMMENTED** [review]
Merged

**Pull request merged**